keylime-agent/src/main.rs

```diff
@@ -586,10 +586,16 @@ async fn main() -> Result<()> {
     let mtls_cert;
     let ssl_context;
     if config.agent.enable_agent_mtls {
+        let contact_ips = vec![config.agent.contact_ip.clone()];
         cert = match config.agent.server_cert.as_ref() {
             "" => {
                 debug!("The server_cert option was not set in the configuration file");
-                crypto::generate_x509(&nk_priv, &agent_uuid)?
+
+                crypto::generate_x509(
+                    &nk_priv,
+                    &agent_uuid,
+                    Some(contact_ips),
```

Comment:
Now that `crypto::generate_x509` is part of the public API, it might make sense to consider switching to the [builder](https://blog.logrocket.com/build-rust-api-builder-pattern/) pattern so we can maintain compatibility more easily in the future:
```rust
crypto::CertBuilder::new(&nk_priv, &agent_uuid)
  .additional_ips(contact_ips)
  .build()?
```

Reply:
Hello @ueno . Thanks for suggestion. I'm a little bit lost with this suggestion ... where is this "CertBuilder" defined? I don't see it anywhere in the code ...

Reply:
Created #755 to track this change